### PR TITLE
Use vault to docker login for pushing client image

### LINKED ISSUE #6513: deployer docker login to push client image regression
[cloud-on-k8s-e2e-tests-ocp/1303/pipeline](https://devops-ci.elastic.co/blue/organizations/jenkins/cloud-on-k8s-e2e-tests-ocp/detail/cloud-on-k8s-e2e-tests-ocp/1303/pipeline)

For kind and ocp, the deployer builds a "client image" with the tool to create the k8s cluster and pushs it in a docker registry.

```
[2023-03-11T00:52:42.545Z] Error response from daemon: manifest for docker.elastic.co/eck-ci/deployer-ocp:4.7.59-791e4cf0a6d5b8ad not found: manifest unknown: manifest unknown
...
[2023-03-11T00:53:01.084Z] #10 naming to docker.elastic.co/eck-ci/deployer-ocp:4.7.59-791e4cf0a6d5b8ad done
[2023-03-11T00:53:01.084Z] #10 DONE 2.1s
[2023-03-11T00:53:01.084Z] 2023/03/11 00:53:00 Not attempting Docker login as .registry.env is not present in the filesystem.
[2023-03-11T00:53:01.084Z] The push refers to repository [docker.elastic.co/eck-ci/deployer-ocp]
[2023-03-11T00:53:01.084Z] 996d5003f1e6: Preparing
[2023-03-11T00:53:01.084Z] denied: requested access to the resource is denied
```

Push fails because the docker client isn't authenticated because docker login was skipped because it relies on the `.registry.env` and #6434 stops writing docker registry credentials to disk.

I missed in #6434 that deployer relies on the `.registry.env`.

https://github.com/elastic/cloud-on-k8s/blob/10ef6a9c8202666c3bf16d05069d59dba48e88f5/hack/deployer/runner/client.go#L80-L89

### PR DESCRIPTION
**Pull request by thbkrkr**
Resolves #6513.

### CODE REVIEW

File: hack/deployer/runner/client.go [hunk 1]
```diff
@@ -77,13 +83,28 @@ func clientImageName(driverID, clientVersion, dockerfileName string) (string, er
...
 
-func dockerLogin() error {
-	registryEnv := ".registry.env"
-	if _, err := os.Stat(registryEnv); os.IsNotExist(err) {
-		// not attempting login when registry env file does not exist (typically outside of CI)
-		log.Printf("Not attempting Docker login as .registry.env is not present in the filesystem.")
-		return nil
+func dockerLogin(client vault.Client) error {
+	// skip docker login in dev if registry exists in docker auth config file
+	if os.Getenv("CI") != "true" {
```
Comment (pebrc) [inline hack/deployer/runner/client.go:88]: Took me a while to figure out that this is set by buildkite. Is this code path no longer used for any Jenkins builds?

**Review by pebrc — APPROVED** [review]
LGTM

File: hack/deployer/runner/client.go (hunk 1 above)
Comment (thbkrkr) [reply hack/deployer/runner/client.go:88]: This code path is used for Buildkite and Jenkins. But setting `CI=true` in the build environment is a standard practice done by almost all known CI systems.

Comment (pebrc) [reply hack/deployer/runner/client.go:88]: TIL!

**Pull request merged**